GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+
+Components that are rendered on both server and client _both_ need to know where to fetch their data from. The values under `DANGEROUSLY_EXPOSE_TO_CLIENT` solve this elegantly. It allows us to have a clear split between universal configuration and secrets that should be server-side only.
+
+`node-config` also supports using individual environmental variables to define important runtime configuration. These can be mapped in the file `custom-environment-variables.json`. See the official  [docs](https://github.com/lorenwest/node-config/wiki/Environment-Variables#custom-environment-variables) for details.
+
+### Reading runtime configuration for your app
+
+So now that we know how to define configuration we need to know how to read and use them in our web application. There are two different use-cases: reading server-only values (including secrets) and reading those available to both server and client (universal).
+
+#### Server only
+
+In your server modules (for example Koa middlewares) you can read runtime configuration like this:
+
+```
+import config from 'config';
+
+console.log(config.host); // github.com
+```
+
+`config` will contain the entire configuration, including secrets.
+
+#### Shared (universal)
+
+In your universal modules (for example React components) you can read runtime configuration like this:
+
+```
+import { appConfig as config } from 'roc-package-web-app-react/app/shared/universal-config';
+
+console.log(config.backendHost); // www.getroc.org
+```
+
+`config` will contain only the configuration that was grouped under `DANGEROUSLY_EXPOSE_TO_CLIENT` in your definition.
+
+## Writing React components
+
+Roc exposes `React` and `prop-types` to your project, and does **not** introduce any abstraction or extra APIs that you need to be aware of. You can define your React components like any other vanilla React project.
+
+Examples:
+
+```js
+import React, { Component } from 'react';
+
+export default class Example extends Component {
+  render() {
+    return (
+      <h1>A component</h1>
+    );
+  }
+}
+```
+
+```js
+import React from 'react';
+
+export default () => (
+  <h1>A stateless component</h1>
+);
+```
+
+You can consult the [official docs for React](https://facebook.github.io/react/docs/components-and-props.html) on components and props.
+
+## Styling React components
+
+Roc does not enforce a particular way of styling components on you. We do however give a fully working styling pipeline as a part of the default template used in this guide; provided by `roc-plugin-style-sass`. This plugin can be found in your `package.json` and removed if you wish to replace it with something else later.
+
+`roc-plugin-style-sass` is a very powerful and flexible way of handling styles in your app. It uses [sass](http://sass-lang.com/) for pre-processing and also includes [css-module](https://github.com/css-modules/css-modules) support out of the box.
```

Comment:
Link to the alternative we have, less?

Reply:
Do we want to advertise that? I am unsure if it is in a suitable state as I have not used it much myself. If it's all good I'll throw in a link to it.